### DISCUSSION TITLE
Incorrect hooks configuration generates incorrect errors

### DISCUSSION THREAD

**Discussion by tyteen4a03** [payloadcms/payload#3662] [Roadmap]
### Link to reproduction

N/A

### To Reproduce

Add this field to a collection:

```
        {
            name: "itemUid",
            type: "text",
            required: true,
            unique: true,
            hooks: {
                beforeValidate: ({data}) => data,
            },
            admin: {
                readOnly: true,
            },
        },
```

### Describe the Bug

The error emitted by this says:

```
[17:18:06] ERROR (payload): There were 1 errors validating your Payload config
[17:18:06] ERROR (payload): 1: Collection "items" > Field "itemUid" > "value" does not match any of the allowed types
```

However, the error is with `hooks.beforeValidate` which does not accept passing a single function (I think this should be changed but not the scope of this bug)

### Payload Version

2.0.5

**Comment by jmikrut** [comment]
Hey @tyteen4a03 — this is actually just the way that Joi generates errors within an `alternatives()` block. It's possible that we can improve it in the future, so we'll add it to the roadmap, but right now this is actually how Joi operates.

I will add this to our roadmap. Thank you for bringing this up as it's certainly misleading!

**Comment by JarrodMFlesch** [comment]
We no longer use joi, TS does not allow this.
